### PR TITLE
[1LP][RFR] New test test_compliance_with_unconditional_policy

### PR DESCRIPTION
**Pull request by quarckster**
{{pytest: -v -k 'test_compliance_with_unconditional_policy'}}

Purpose or Intent
==============

New test test_compliance_with_unconditional_policy https://bugzilla.redhat.com/show_bug.cgi?id=1369488

### CODE REVIEW

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/control/explorer/policy_profiles.py [hunk 1]
```diff
@@ -178,5 +178,4 @@ class PolicyProfileDetails(CFMENavigateStep):
     prerequisite = NavigateToAttribute("appliance.server", "ControlExplorer")
 
     def step(self):
-        if self.view.policy_profiles.is_opened:
```
Comment (psav) [inline cfme/control/explorer/policy_profiles.py]: Why drop the check? unless this is now a part of click_path?

File: cfme/infrastructure/host.py [hunk 2]
```diff
@@ -214,6 +217,12 @@ def delete(self, cancel=True):
         cfg_btn(btn_name, invokes_alert=True)
         sel.handle_alert(cancel=cancel)
 
+    def load_details(self, refresh=False):
+        """To be compatible with the Taggable and PolicyProfileAssignable mixins."""
+        navigate_to(self, 'Details')
```
Comment (psav) [inline cfme/infrastructure/host.py]: Thinking of adding a refresh option to all pages, like the use_resetter.....thoughts? @mfalesni @RonnyPfannschmidt @jkrocil @izapolsk @mshriver @otsuman

File: cfme/infrastructure/host.py [hunk 3]
```diff
@@ -329,6 +303,66 @@ def run_smartstate_analysis(self):
...
+
+        Note:
+            The host must have assigned policy profile with host compliance policy.
+        """
+        navigate_to(self, 'Details')
+        tb.select('Policy', 'Check Compliance of Last Known Configuration', invokes_alert=True)
+        sel.handle_alert()
+        flash.assert_no_errors()
+
+    @contextmanager
```
Comment (psav) [inline cfme/infrastructure/host.py]: This is interesting idea and I kinda like it, though I think it is a methodology that we should explore fully before we start using it. Thoughts ?  @mfalesni @RonnyPfannschmidt @jkrocil @izapolsk @mshriver @otsuman

File: cfme/control/explorer/policy_profiles.py (hunk 1 above)
Comment (quarckster) [inline cfme/control/explorer/policy_profiles.py]: It doesn't work for me with this check

Comment (mfalesni) [inline cfme/control/explorer/policy_profiles.py]: accordions open automatically when child widgets are accessed

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by quarckster** [comment]
@jkrocil please review

File: cfme/infrastructure/host.py (hunk 2 above)
Comment (jkrocil) [inline cfme/infrastructure/host.py]: Nope, don't like it. Both `check_compliance` and `check_compliance_wrapper` should be removed not only from this PR but also from `cfme/common/vm.py`. It is 2 year old code that is not used anywhere.

Comment (jkrocil) [inline cfme/infrastructure/host.py]: It definitely needs some refactoring but bottom-line - we don't need 3 functions to do one thing.

Comment (jkrocil) [inline cfme/infrastructure/host.py]: All this was meant for the `@contextmanager` stuff below, sorry!

**Comment by dajoRH** [comment]
Lint report for commit 1bdba4bc460d21dd43817d97077296aab6a56ae4:

`cfme/infrastructure/host.py`:
- :red_circle: Line 11:1: **F401** *'contextlib.contextmanager' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/infrastructure/host.py [hunk 4]
```diff
@@ -329,6 +302,47 @@ def run_smartstate_analysis(self):
...
+        compared.
+
+        Returns:
+            :py:class:`NoneType` if no title is present (no compliance checks before), otherwise str
+        """
+        sel.refresh()
+        return self.get_detail('Compliance', 'Status')
+
+    @property
+    def compliant(self):
```
Comment (jkrocil) [inline cfme/infrastructure/host.py]: This should be `is_compliant`.

File: cfme/infrastructure/host.py [hunk 5]
```diff
@@ -329,6 +302,47 @@ def run_smartstate_analysis(self):
...
+        navigate_to(self, 'Details')
+        original_state = self.compliance_status
+        tb.select('Policy', 'Check Compliance of Last Known Configuration', invokes_alert=True)
+        sel.handle_alert()
+        flash.assert_no_errors()
+        wait_for(
+            lambda: self.compliance_status != original_state,
+            num_sec=timeout, delay=5, message="compliance of {} checked".format(self.name)
+        )
+        return self.compliant
```
Comment (jkrocil) [inline cfme/infrastructure/host.py]: I don't think this should return current compliance status. It should only ensure that the appropriate button is clicked and wait for the compliance check to finish. If we want to assure that the host is compliant after running this, we should do `assert host.is_compliant`.

File: cfme/tests/control/test_compliance.py [hunk 6]
```diff
@@ -203,3 +235,12 @@ def test_check_files(request, fleecing_vm, ssh_client, analysis_profile):
 
     do_scan(fleecing_vm, ("Configuration", "Files"))
     assert fleecing_vm.check_compliance()
```
Comment (jkrocil) [inline cfme/tests/control/test_compliance.py]: This needs to be updated as well.

Comment (jkrocil) [inline cfme/tests/control/test_compliance.py]: That's a VM, my bad.

**Pull request merged**